packages/utils/src/cache/cache-indexeddb.ts

```diff
@@ -0,0 +1,161 @@
+import { openDB } from 'idb';
+import type { DBSchema, IDBPDatabase } from 'idb';
+import type {
+  CacheStorage,
+  CacheItem,
+  CacheMap,
+  CacheIndexedDBStorageConfig
+} from './types';
+
+interface IndexedDBDatabase extends DBSchema {
+  [key: string]: {
+    key: string;
+    value: CacheItem;
+  }
+}
+
+const isFirefox = () => (/firefox/i).test(navigator.userAgent);
+
+/**
+ * Stores data in indexedDB for use across multiple sessions.
+ */
+export class CacheIndexedDBStorage implements CacheStorage {
+
+  /**
+   * A connection to a indexedDB database, use for open transaction or idb api
+   */
+  public db: IDBPDatabase<IndexedDBDatabase> | undefined;
+
+  /**
+   * Database name
+   */
+  public dbName: string;
+
+  /**
+   * Database Version.
+   */
+  public version: number;
+
+  /**
+   * Store name
+   */
+  public storeName: never;
+
+  /**
+   * Constructor
+   * @param info Storage config
+   */
+  constructor (info: CacheIndexedDBStorageConfig) {
+    const { dbName, version, storeName } = info;
+    this.dbName = dbName;
+    this.version = version;
+    this.storeName = storeName as never;
+  }
+
+  /**
+   * Set a item against a key to this storage
+   * @param key item key
+   * @param value item value
+   * @returns {void}
+   */
+  public async setItem (key: string, value: CacheItem): Promise<void> {
+    !this.db && await this.open();
```

Comment:
I see `!this.db && await this.open();` everywhere, cloud you create a new method and move to it?

Reply:
Due to [Trem feedback](https://github.com/Refinitiv/refinitiv-ui/pull/412#pullrequestreview-1066797912), ready logic should move to each storage class. So the line will be remove and use this.ready instead.